Panics in latest version

Issue started happening in v1.2.9 (could not duplicate in v1.2.8). It's hard to point out where it's happening but seems to be related using `ScanAndCount` method with `Relation` in the query.

```
goroutine 30 [running]:
github.com/uptrace/bun.(*whereBaseQuery).appendWhere(0x14000421680, {{0x103431a38?, 0x14000132858?}, 0x1400000c930?}, {0x140000fc000?, 0x32?, 0x40?}, 0x1)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_base.go:832 +0x3b0
github.com/uptrace/bun.(*SelectQuery).appendQuery(0x14000421680, {{0x103431a38?, 0x14000132858?}, 0x0?}, {0x140000fc000, 0x2aa, 0x380}, 0x1)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:605 +0x7dc
github.com/uptrace/bun.countQuery.AppendQuery({0x14000384f18?}, {{0x103431a38?, 0x14000132858?}, 0x0?}, {0x0?, 0x4?, 0x14000384f38?})
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:1176 +0x58
github.com/uptrace/bun.(*SelectQuery).Count(0x14000421680, {0x10342ec10, 0x1038069c0})
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:929 +0x58
github.com/uptrace/bun.(*SelectQuery).scanAndCountConcurrently.func2()
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:990 +0x6c
created by github.com/uptrace/bun.(*SelectQuery).scanAndCountConcurrently in goroutine 28
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:986 +0x1d8
exit status 2
FAIL    github.com/----/postgres      6.638s

```

Here is once piece of code that leads to this failure:

```go
count, err := m.DB.NewSelect().
	Model(&attachments).
	Relation("User", func(q *bun.SelectQuery) *bun.SelectQuery {
		return q.ExcludeColumn("*")
	}).
        ScanAndCount(ctx)

```

I will get different panic traces each time I run my test suite, here is another for reference:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10501d0b0]

goroutine 74 [running]:
github.com/uptrace/bun.(*baseQuery).addColumn(...)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_base.go:353
github.com/uptrace/bun.(*SelectQuery).Column(...)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:118
github.com/----/postgres.(*Modeler).AttachmentsByQuery.func1(0x140004c5680)
        /----/postgres/attachment.go:93 +0xf4
github.com/uptrace/bun.(*SelectQuery).Relation.(*SelectQuery).applyToRelation.func2(0x0?)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:442 +0x44
github.com/uptrace/bun.(*relationJoin).applyTo(0x140003c5ba0, 0x140004c5680)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/relation_join.go:37 +0xb4
github.com/uptrace/bun.(*SelectQuery).appendQuery.func1(0x0?)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:540 +0x20
github.com/uptrace/bun.(*SelectQuery)._forEachInlineRelJoin(0x140004c5680, 0x140004bed68, {0x140003c5ba0, 0x4, 0x0?})
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:461 +0x70
github.com/uptrace/bun.(*SelectQuery).forEachInlineRelJoin(0x140004c5680, 0x140004bed68)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:453 +0x50
github.com/uptrace/bun.(*SelectQuery).appendQuery(0x140004c5680, {{0x10559da38?, 0x14000132858?}, 0x0?}, {0x14000091000, 0x0, 0x1000}, 0x0)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:539 +0x210
github.com/uptrace/bun.(*SelectQuery).AppendQuery(0x140004c5680, {{0x10559da38?, 0x14000132858?}, 0x0?}, {0x14000091000?, 0x0?, 0x1400005d6a8?})
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:513 +0x7c
github.com/uptrace/bun.(*SelectQuery).scanResult(0x140004c5680, {0x10559ac10, 0x1059729c0}, {0x0?, 0x0, 0x0?})
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:875 +0x1a8
github.com/uptrace/bun.(*SelectQuery).Scan(...)
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:843
github.com/uptrace/bun.(*SelectQuery).scanAndCountConcurrently.func1()
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:975 +0x70
created by github.com/uptrace/bun.(*SelectQuery).scanAndCountConcurrently in goroutine 73
        /go/pkg/mod/github.com/uptrace/bun@v1.2.9/query_select.go:972 +0x134
exit status 2
FAIL    github.com/----/postgres      6.801s
```

This one was referencing this line in my code:

```
    Relation("VerifiedBy", func(q *bun.SelectQuery) *bun.SelectQuery {
        return q.Column("id", "first_name", "last_name") // <---
    }).

```

So seems to be related to `ScanAndCount` and `Relation` somehow.

--- a/query_select.go
+++ b/query_select.go
@@ -967,27 +967,27 @@ func (q *SelectQuery) scanAndCountConcurrently(
 	var mu sync.Mutex
 	var firstErr error
 
-	if q.limit >= 0 {
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
+	countQuery := q.Clone()
 
-			if err := q.Scan(ctx, dest...); err != nil {
-				mu.Lock()
-				if firstErr == nil {
-					firstErr = err
-				}
-				mu.Unlock()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		if err := q.Scan(ctx, dest...); err != nil {
+			mu.Lock()
+			if firstErr == nil {
+				firstErr = err
 			}
-		}()
-	}
+			mu.Unlock()
+		}
+	}()
 
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
 
 		var err error
-		count, err = q.Count(ctx)
+		count, err = countQuery.Count(ctx)
 		if err != nil {
 			mu.Lock()
 			if firstErr == nil {
@@ -1075,6 +1075,120 @@ func (q *SelectQuery) String() string {
 	}
 
 	return string(buf)
+}
+
+func (q *SelectQuery) Clone() *SelectQuery {
+	if q == nil {
+		return nil
+	}
+
+	cloneArgs := func(args []schema.QueryWithArgs) []schema.QueryWithArgs {
+		if len(args) == 0 {
+			return nil
+		}
+		clone := make([]schema.QueryWithArgs, len(args))
+		copy(clone, args)
+		return clone
+	}
+	cloneHints := func(hints *indexHints) *indexHints {
+		if hints == nil {
+			return nil
+		}
+		return &indexHints{
+			names:      cloneArgs(hints.names),
+			forJoin:    cloneArgs(hints.forJoin),
+			forOrderBy: cloneArgs(hints.forOrderBy),
+			forGroupBy: cloneArgs(hints.forGroupBy),
+		}
+	}
+
+	clone := &SelectQuery{
+		whereBaseQuery: whereBaseQuery{
+			baseQuery: baseQuery{
+				db:             q.db,
+				table:          q.table,
+				model:          q.model,
+				tableModel:     q.tableModel,
+				with:           make([]withQuery, len(q.with)),
+				tables:         cloneArgs(q.tables),
+				columns:        cloneArgs(q.columns),
+				modelTableName: q.modelTableName,
+			},
+			where: make([]schema.QueryWithSep, len(q.where)),
+		},
+
+		idxHintsQuery: idxHintsQuery{
+			use:    cloneHints(q.idxHintsQuery.use),
+			ignore: cloneHints(q.idxHintsQuery.ignore),
+			force:  cloneHints(q.idxHintsQuery.force),
+		},
+
+		orderLimitOffsetQuery: orderLimitOffsetQuery{
+			order:  cloneArgs(q.order),
+			limit:  q.limit,
+			offset: q.offset,
+		},
+
+		distinctOn: cloneArgs(q.distinctOn),
+		joins:      make([]joinQuery, len(q.joins)),
+		group:      cloneArgs(q.group),
+		having:     cloneArgs(q.having),
+		union:      make([]union, len(q.union)),
+		comment:    q.comment,
+	}
+
+	for i, w := range q.with {
+		clone.with[i] = withQuery{
+			name:      w.name,
+			recursive: w.recursive,
+			query:     w.query, // TODO: maybe clone is need
+		}
+	}
+
+	if !q.modelTableName.IsZero() {
+		clone.modelTableName = schema.SafeQuery(
+			q.modelTableName.Query,
+			append([]any(nil), q.modelTableName.Args...),
+		)
+	}
+
+	for i, w := range q.where {
+		clone.where[i] = schema.SafeQueryWithSep(
+			w.Query,
+			append([]any(nil), w.Args...),
+			w.Sep,
+		)
+	}
+
+	for i, j := range q.joins {
+		clone.joins[i] = joinQuery{
+			join: schema.SafeQuery(j.join.Query, append([]any(nil), j.join.Args...)),
+			on:   make([]schema.QueryWithSep, len(j.on)),
+		}
+		for k, on := range j.on {
+			clone.joins[i].on[k] = schema.SafeQueryWithSep(
+				on.Query,
+				append([]any(nil), on.Args...),
+				on.Sep,
+			)
+		}
+	}
+
+	for i, u := range q.union {
+		clone.union[i] = union{
+			expr:  u.expr,
+			query: u.query.Clone(),
+		}
+	}
+
+	if !q.selFor.IsZero() {
+		clone.selFor = schema.SafeQuery(
+			q.selFor.Query,
+			append([]any(nil), q.selFor.Args...),
+		)
+	}
+
+	return clone
 }
 
 //------------------------------------------------------------------------------

--- a/schema/sqlfmt.go
+++ b/schema/sqlfmt.go
@@ -50,6 +50,7 @@ func (s Ident) AppendQuery(fmter Formatter, b []byte) ([]byte, error) {
 
 //------------------------------------------------------------------------------
 
+// NOTE: It should not be modified after creation.
 type QueryWithArgs struct {
 	Query string
 	Args  []interface{}

